packages/eslint-plugin/src/utils/runtime-lint-utils.ts

```diff
@@ -490,35 +499,57 @@ export function groupImports(
 /**
  * Checks if source file belongs to a secondary entry point different than the import one
  */
-export function belongsToDifferentNgEntryPoint(
+export function belongsToDifferentEntryPoint(
   importExpr: string,
   filePath: string,
   projectRoot: string
 ): boolean {
+  const importEntryPoint = getSecondaryEntryPointPath(
+    importExpr,
+    filePath,
+    projectRoot
+  );
+  const srcEntryPoint = getEntryPoint(filePath, projectRoot);
+
+  // check if the entry point of import expression is different than the source file's entry point
+  return importEntryPoint !== srcEntryPoint;
+}
+
+export function getSecondaryEntryPointPath(
+  importExpr: string,
+  filePath: string,
+  projectRoot: string
+): string | undefined {
   const resolvedImportFile = resolveModuleByImport(
     importExpr,
     filePath, // not strictly necessary, but speeds up resolution
     path.join(workspaceRoot, getRootTsConfigFileName())
   );
-
   if (!resolvedImportFile) {
-    return false;
+    return undefined;
   }
+  const entryPoint = getEntryPoint(resolvedImportFile, projectRoot);
+  return entryPoint;
+}
 
-  const importEntryPoint = getAngularEntryPoint(
-    resolvedImportFile,
-    projectRoot
+function getEntryPoint(file: string, projectRoot: string): string {
+  const packageEntryPoints = getPackageEntryPoints(projectRoot);
+  const fileEntryPoint = packageEntryPoints.find(
```

Comment:
The entry points could have patterns (`*`), which would require a more complex matching to find whether an import matches a given entry point.

Reply:
Let's handle this later if users request it. I'm worried this might impact performance significantly so I would not add it unless requested.